spring-boot-starter/starter-client/spring/src/main/java/org/camunda/bpm/client/spring/impl/subscription/SubscriptionPostProcessor.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.client.spring.impl.subscription;
+
+import org.camunda.bpm.client.spring.impl.client.util.ClientLoggerUtil;
+import org.camunda.bpm.client.spring.annotation.ExternalTaskSubscription;
+import org.camunda.bpm.client.spring.impl.subscription.util.SubscriptionLoggerUtil;
+import org.camunda.bpm.client.spring.impl.util.AnnotationUtil;
+import org.camunda.bpm.client.task.ExternalTaskHandler;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.ListableBeanFactory;
+import org.springframework.beans.factory.annotation.AnnotatedBeanDefinition;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanDefinitionRegistryPostProcessor;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+public class SubscriptionPostProcessor implements BeanDefinitionRegistryPostProcessor {
```

Comment:
```suggestion
public class SubscriptionPostProcessor implements ImportBeanDefinitionRegistrar {
```
I think we should use `ImportBeanDefinitionRegistrar` instead of `BeanDefinitionRegistryPostProcessor`. While it seems to work, `BeanDefinitionRegistryPostProcessor` is not meant to be imported via annotation but rather exposed as a dedicated bean picked up by spring.

Reply:
That sounds reasonable 👍

Reply:
That sounds reasonable 👍

Reply:
For the record, we decided the following:
* Changing the `BeanDefinitionRegistryPostProcessor` to `ImportBeanDefinitionRegistrar` does not work since the latter has only class-level beans in the bean definition registry and not method-level beans which is a requirement to register subscription beans in a sensible way
* Since`BeanDefinitionRegistryPostProcessor` is more powerful and more commonly used compared to `ImportBeanDefinitionRegistrar`, we decided to convert the `ClientRegistry` into a `BeanDefinitionRegistryPostProcessor`
* Both post processors are registered at a central point in `PostProcessorConfiguration` and imported via `@Import` annotation in the `@EnableExternalTaskClient` annotation
  * Whenever the `@EnableExternalTaskClient` is added to a class, the `ClientPostProcessor` as well as `SubscriptionPostProcessor` are activated

Reply:
Documented in CAM-13254